ui/app/components/currency-display/currency-display.container.js

```diff
@@ -20,15 +20,28 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => {
     currency,
     denomination,
     hideLabel,
+    displayValue: propsDisplayValue,
+    suffix: propsSuffix,
     ...restOwnProps
   } = ownProps
 
   const toCurrency = currency || currentCurrency
-  const convertedValue = getValueFromWeiHex({
-    value, fromCurrency: nativeCurrency, toCurrency, conversionRate, numberOfDecimals, toDenomination: denomination,
-  })
-  const displayValue = formatCurrency(convertedValue, toCurrency)
-  const suffix = hideLabel ? undefined : toCurrency.toUpperCase()
+
+  let displayValue = propsDisplayValue
+
+  if (typeof displayValue === 'undefined') {
+    const convertedValue = getValueFromWeiHex({
+      value,
+      fromCurrency: nativeCurrency,
+      toCurrency, conversionRate,
+      numberOfDecimals,
+      toDenomination: denomination,
+    })
+
+    displayValue = propsDisplayValue || formatCurrency(convertedValue, toCurrency)
```

Comment:
If we inline `convertedValue` could we write this as a single `cost displayValue` statement without the conditional?